aten/src/THC/THCAllocator.cpp

```diff
@@ -4,21 +4,30 @@ THCIpcDeleter::~THCIpcDeleter() {
   int prev_device;
   THCudaCheck(cudaGetDevice(&prev_device));
   THCudaCheck(cudaSetDevice(device_));
-  THCudaCheck(cudaIpcCloseMemHandle(data_));
   THCudaCheck(cudaSetDevice(prev_device));
 }
 
 void deleteTHCIpcDeleter(void* ptr) {
   delete static_cast<THCIpcDeleter*>(ptr);
 }
 
-at::DataPtr THCIpcDeleter::makeDataPtr(void* data, int device) {
+// Refer to NB [CUDA IPC and the caching allocator] for more details
+// basePtr - device ptr of allocated CUDA memory region. This memory region
+//           might contain storages of different types.
+// data    - ptr to where the storage (of a single type) should start.
+// device  - device of memory
+// Here basePtr should be saved in the struct, while data should be used to
+// construct the new storage.
+// Every time a storage on the memory region go out of scope, the ref_count
```

Comment:
nit: Well, it's whatever the shared pointers internal refcount is; ref_count seems to imply that we're manually refcounting something by the name of the ref_count identifier but we're not

Reply:
grammar: Every time a storage referring to the IPC memory region goes out of scope, the reference count on the memory region will be decreased by one, until it's zero, at which point IPC memory region is closed (by calling cudaIpcCloseMemHandle).